code/game/objects/structures/tables_racks.dm

```diff
@@ -113,14 +113,12 @@
 	if(!ishuman(pushed_mob))
 		return
 	var/mob/living/carbon/human/H = pushed_mob
-	GET_COMPONENT_FROM(mood, /datum/component/mood, H)
-	if(mood)
-		if(iscatperson(H)) //Catpeople are a bit dumb and think its fun to be on a table
-			mood.add_event("table", /datum/mood_event/happytable)
-			H.startTailWag()
-			addtimer(CALLBACK(H, /mob/living/carbon/human.proc/endTailWag), 30)
-		else
-			mood.add_event("table", /datum/mood_event/table)
+	if(iscatperson(H)) //Catpeople are a bit dumb and think its fun to be on a table

```

Comment:
I feel like this should be handled by the component, can it not look at the owner of the signal to decide if the event is good or bad instead of having to code it in the sending code?